datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt

```diff
@@ -418,9 +439,40 @@ internal class QuestionnaireViewModel(application: Application, state: SavedStat
           )
           .map { it.copy() }
       unpackRepeatedGroups(this@QuestionnaireViewModel.questionnaire)
+      addTimeStampValue(item, "submission-timestamp")
     }
   }
 
+    private fun addTimeStampValue(items: List<QuestionnaireResponseItemComponent>, itemLinkId : String) {
+        if (shouldAddTimestampItems) {
+            items.forEach { item ->
+                if (item.linkId == itemLinkId) {
+                    item.answer = arrayListOf<QuestionnaireResponseItemAnswerComponent?>().apply {
+                        add(QuestionnaireResponseItemAnswerComponent().apply {
+                            value = DateTimeType(Date().toTimeZoneString())
+                        })
+                    }
+                }
+            }
+        }
+    }
+
+    private fun Date.toTimeZoneString(): String {
+        val simpleDateFormat =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", Locale.getDefault())
+                .withZone(ZoneId.systemDefault())
+        return simpleDateFormat.format(this.toInstant())
+    }
+
+    private fun getTimestampItem(linkIdVal : String): QuestionnaireItemComponent? = if (shouldAddTimestampItems)
+        QuestionnaireItemComponent().apply {
+            linkId = linkIdVal
+            readOnly = true
+            type = Questionnaire.QuestionnaireItemType.DATETIME
+            extension.add(Extension("http://hl7.org/fhir/StructureDefinition/questionnaire-hidden", BooleanType(true)))
+        } else
+          null
+
```

Comment:
Based on the previous comments

```suggestion
    private fun getTimestampItem(linkIdVal : String) = 
        QuestionnaireItemComponent().apply {
            linkId = linkIdVal
            readOnly = true
            type = Questionnaire.QuestionnaireItemType.DATETIME
            extension.add(Extension("http://hl7.org/fhir/StructureDefinition/questionnaire-hidden", BooleanType(true)))
        } 

```